### PR TITLE
Feature/syal 925

### PR DESCRIPTION
**Pull request by HieuLM-AVISPL**
implement create/delete stream control

### CODE REVIEW

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/ControllingMetricGroup.java [hunk 1]
```diff
@@ -12,42 +15,54 @@
...
 	 */
-	ControllingMetricGroup(String name) {
-		this.name = name;
+	ControllingMetricGroup(String uiName, String apiName) {
+		this.uiName = uiName;
+		this.apiName = apiName;
+	}
 
+	/**
+	 * retrieve {@code {@link #uiName }}
```
Comment (ThanhD-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/ControllingMetricGroup.java]: retrieve => Retrieves

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/ErrorMessage.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	 * @param errorMessage error message that want to convert
+	 * @return String converted error message
+	 */
+	public static String convertErrorMessage(String errorMessage) {
+		if (errorMessages == null) {
+			errorMessages = new HashMap<>();
+			errorMessages.put("Decoder 2 configuration failed: \\\"Stream still in use\\\"\"}", "Please choose a different secondary stream");
+		}
+		Optional<String> result = Optional.ofNullable(errorMessages.get(errorMessage));
+		if (result.isPresent()) {
```
Comment (ThanhD-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/ErrorMessage.java]: Convert this if to return result.orElse(errorMessage);

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/dto/streamstats/StreamConfig.java [hunk 3]
```diff
@@ -417,6 +423,125 @@ public String getDefaultStreamName() {
...
+		request.append(command.concat(DecoderConstant.SPACE)
+				.concat(action));
+
+		Encapsulation encapsulationEnum = Encapsulation.getByApiName(getDefaultValueForNullData(this.encapsulation, DecoderConstant.EMPTY));
+		SRTMode srtModeEnum = SRTMode.getByName(getDefaultValueForNullData(this.srtMode, DecoderConstant.EMPTY));
+		SwitchOnOffControl aeEncryptedEnum = SwitchOnOffControl.getByName(getDefaultValueForNullData(this.srtSettings, DecoderConstant.EMPTY));
+		SwitchOnOffControl streamFlippingEnum = SwitchOnOffControl.getByName(getDefaultValueForNullData(this.streamFlipping, DecoderConstant.EMPTY));
+
+		if (!StringUtils.isNullOrEmpty(name)) {
+			request.append(" name=\"" + name + DecoderConstant.DOUBLE_QUOTATION);
```
Comment (ThanhD-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/dto/streamstats/StreamConfig.java]: Convert request.append(" name=\"" + name + DecoderConstant.DOUBLE_QUOTATION); to
request.append(" name=\"").append(name).append(DecoderConstant.DOUBLE_QUOTATION);
Same for the rest of append.

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java [hunk 4]
```diff
@@ -2190,6 +2312,21 @@ public void createStreamControl(Map<String, String> stats, List<AdvancedControll
 		}
 	}
 
+	/**
+	 * This method is used to perform create stream control
+	 */
```
Comment (ThanhD-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java:2326]: Add throws ResourceNotReachableException in javadoc.

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java [hunk 5]
```diff
@@ -1579,45 +1615,76 @@ private void populateCreateStreamControl(Map<String, String> stats, List<Advance
...
-		List<String> fecRTPs = DropdownList.getListOfEnumNames(Fec.class);
+		Fec fecEnum = Fec.getByAPIStatsName(getDefaultValueForNullData(cachedStreamConfig.getFec(), DecoderConstant.EMPTY));
+		Fec fecEnumCopy = Fec.DISABLE;
+
+		// Get list values of controllable property (dropdown list)
+		List<String> fecModes = DropdownList.getListOfEnumNames(Fec.class, Fec.VF);
+
+		for (String fecElement : fecModes) {
+			if (fecElement.equals(fecEnum.getUiName())) {
+				fecEnumCopy = fecEnum;
```
Comment (ThanhD-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java:1680]: Add break;

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/ErrorMessage.java (hunk 2 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/ErrorMessage.java]: I fixed it

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/ControllingMetricGroup.java (hunk 1 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/ControllingMetricGroup.java]: I fixed it

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/dto/streamstats/StreamConfig.java (hunk 3 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/dto/streamstats/StreamConfig.java]: I fixed it

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java (hunk 4 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java:2326]: I fixed it

File: src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java (hunk 5 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java:1680]: I fixed it

**Pull request merged**